tests/integration/ivaDefects.intTest.ts

```diff
@@ -1,79 +1,48 @@
+        const expectedResponse = JSON.parse(JSON.stringify(ivaDefectsData)).map(
+          (defect: { id: any }) => {
             delete defect.id;
             return defect;
-          })
-          .sort(
-            (first: { imNumber: number }, second: { imNumber: number }) =>
-              first.imNumber - second.imNumber,
-          );
-
+          },
+        );
         await request
-          .get("defects/iva/manual/M1")
+          .get("defects/iva?euVehicleCategory=m1")
           .set({ Authorization: mockToken })
           .then((res: any) => {
             expect(res.statusCode).toBe(200);
             expect(res.headers["access-control-allow-origin"]).toBe("*");
             expect(res.headers["access-control-allow-credentials"]).toBe(
               "true",
             );
-            expect(res.body.length).toBe(expectedResponse.length);
-          });
-      });
 
-      it("should return an empty array for unknown id", async () => {
-        const expectedResponse = [];
-
-        await request
-          .get("defects/iva/manual/fakeId")
-          .set({ Authorization: mockToken })
-          .then((res: any) => {
-            expect(res.statusCode).toBe(204);
-            expect(res.headers["access-control-allow-origin"]).toBe("*");
-            expect(res.headers["access-control-allow-credentials"]).toBe(
-              "true",
-            );
-            expect(res.body.length).toBe(expectedResponse.length);
+            expect(res.body).not.toBeNull();
+            expect(res.body?.euVehicleCategories?.at(0)).toBe("m1");
+            expect(res.body?.basic?.length).toBe(49);
+            expect(res.body?.normal?.length).toBe(51);
           });
       });
-    });
-  });
-
-  describe("getIvaDefects", () => {
-    context("when database is populated", () => {
-      it("should return all defects in the database", async () => {
-        const expectedResponse = JSON.parse(JSON.stringify(ivaDefectsData))
-          .map((defect: { id: any }) => {
-            delete defect.id;
-            return defect;
-          })
-          .sort(
-            (first: { imNumber: number }, second: { imNumber: number }) =>
-              first.imNumber - second.imNumber,
-          );
 
+      it("should return an empty response for an unknown euVehicleCategory", async () => {
         await request
-          .get("defects/iva")
+          .get("defects/iva?euVehicleCategory=f1")
           .set({ Authorization: mockToken })
           .then((res: any) => {
-            expect(res.statusCode).toBe(200);
+            expect(res.statusCode).toBe(204);
             expect(res.headers["access-control-allow-origin"]).toBe("*");
             expect(res.headers["access-control-allow-credentials"]).toBe(
               "true",
             );
-            expect(res.body.length).toBe(expectedResponse.length);
+            expect(res.body).toBe("");
```

Comment:
@LGin-BJSS This is where I was referencing, from the function I would have expected this to be "[ ]" rather than "" - the logic for the function is below.

```
  if (!euVehicleCategoryQuery) {
    return new HTTPResponse(400, "euVehicleCategory required");
  } else if (
    !Object.values(EUVehicleCategory).includes(euVehicleCategoryQuery)
  ) {
    return new HTTPResponse(204, "[]");
  } 
```

Reply:
Actually, thinking about it. Is it actually possible to return a body for a 204 status?

Looking at the spec: 

**_The 204 (No Content) status code indicates that the server has successfully fulfilled the request and that there is no additional content to send in the response payload body._**

Reply:
its definitely possible - although, not strictly as intended by the RFC.

Reply:
I stand corrected, updating the code to 200, returns the empty array - so something in the framework is actively terminating the response before the body.

Reply:
Yes when I was doing a local debug, that is what drove me to check the Http Status Code specs.

```
The 204 response MUST NOT include a message-body, and thus is always terminated by the first empty line after the header fields. 
```